tests/kafkatest/sanity_checks/test_verifiable_producer.py

```diff
@@ -0,0 +1,70 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from ducktape.tests.test import Test
+from ducktape.utils.util import wait_until
+from ducktape.mark import parametrize
+
+from kafkatest.services.zookeeper import ZookeeperService
+from kafkatest.services.kafka import KafkaService
+from kafkatest.services.kafka.version import LATEST_0_8_2, TRUNK, KafkaVersion
+from kafkatest.services.verifiable_producer import VerifiableProducer
+from kafkatest.utils import is_version
+
+
+class TestVerifiableProducer(Test):
+    """Sanity checks on verifiable producer service class."""
+    def __init__(self, test_context):
+        super(TestVerifiableProducer, self).__init__(test_context)
+
+        self.topic = "topic"
+        self.zk = ZookeeperService(test_context, num_nodes=1)
+        self.kafka = KafkaService(test_context, num_nodes=1, zk=self.zk,
+                                  topics={self.topic: {"partitions": 1, "replication-factor": 1}})
+
+        self.num_messages = 100
+        # This will produce to source kafka cluster
+        self.producer = VerifiableProducer(test_context, num_nodes=1, kafka=self.kafka, topic=self.topic,
+                                           max_messages=self.num_messages, throughput=1000)
+
+    def setUp(self):
+        self.zk.start()
+        self.kafka.start()
+
+    @parametrize(version=str(LATEST_0_8_2))
```

Comment:
Hmm, I'm not sure that these have to all validate the same things in the same way; for example, in this case with versioned VerfiableProducer, my primary concern was that the process actually started, given the classpath tricks we have to play to run with 0.8.x producers. This is not a bulletproof test or anything, but it catches major problems **quickly**.

Yes, the "validation" is just a crude heuristic, but it tightens the development feedback loop! Among other things, it provided a signal when the classpath was broken, and immediately revealed that rebasing brought in a dependency on a method on `kafka.common` not present in 0.8.X, therefore breaking the way VerifiableProducer is run using a mix of 0.8 and 0.9 jars.

I can't speak for other test developers, but I have found there is *tremendous* value in finding out **quickly** that some minor changes to a remote start command did not suddenly cause the service to fail to start.

With respect to optimization, these aren't really the long pole in the tent - less than 5 minutes out of 3 hours as of last night - so if I were to optimize overall test run time, this is not where I'd focus :)

It's possible that some of these tests are really only useful during more active development phase and can be discarded over time.